inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -105,7 +105,7 @@ def add(self, parameters: ApplicationAddSourceParameters) -> None:
         # Step 2: Add the source
         try:
             create_file_with_contents(
-                os.path.join(UBUNTU_APT_SOURCES_LIST_D, parameters.file_name), parameters.sources
+                os.path.join(UBUNTU_APT_SOURCES_LIST_D, parameters.source_list_file_name), parameters.sources
```

Comment:
Validate the `source_list_file_name` to ensure it does not contain any directory traversal characters before using it in file paths. [important]